src/notfoundigest/handler-external.js

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2024 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import { internalServerError, noContent } from '@adobe/spacecat-shared-http-utils';
+import { isObject } from '@adobe/spacecat-shared-utils';
+import { build404SlackMessage, post404InitialSlackMessage, postSlackMessage } from '../support/slack.js';
+import { get404Backlink } from '../support/utils.js';
+
+const ALERT_TYPE = '404';
+
+export default async function notFoundExternalDigestHandler(message, context) {
+  const { dataAccess, log } = context;
+  const {
+    env: {
+      SLACK_BOT_TOKEN: token,
+    },
+  } = context;
+
+  const organizations = await dataAccess.getOrganizations();
+  for (const organization of organizations) {
+    const orgConfig = organization.getConfig();
+
+    const notFoundOrgAlertConfig = orgConfig.alerts.find((alert) => alert.type === ALERT_TYPE);
+    const organizationId = organization.getId();
+    // eslint-disable-next-line no-await-in-loop
+    const sites = await dataAccess.getSitesByOrganizationID(organizationId);
```

Comment:
here, `dataAccess.getSitesByOrganizationID` is called. Then later on, `dataAccess.getLatestAuditForSite` is called for each site.

we could add a new method `getSitesWithLatestAuditByOrganizationId` to data-access; which would eliminate all these complexity plus n + 1 db calls